### ISSUE TITLE
Seg fault on v0.19 beta

### ISSUE THREAD

**Issue by Nick-Mazuk** [typesense/typesense#212]
## Description

Segfault on the beta version of Typesense.

## Steps to reproduce

Unfortunately, all I have is the terminal output. It crashed in the background as I was working on something else. My guess is that it crashed during a hot refresh of my website on localhost.

## Expected Behavior

No seg fault, which crashes Typesense.

## Actual Behavior

```bash
ackward.hpp:4192] Stack trace (most recent call last) in thread 108:
#11   Object "", at 0xffffffffffffffff, in 
#10   Object "/lib/x86_64-linux-gnu/libc-2.23.so", at 0x7fef308b241c, in __clone
#90   Object "/lib/x86_64-linux-gnu/libpthread-2.23.so", at 0x7fef310896b9, in start_thread
#80   Source "../../../../../libstdc++-v3/src/c++11/thread.cc", line 83, in execute_native_thread_routine [0x134452e]
#70   Source "/typesense/src/index.cpp", line 1227, in run_search [0x90171f]
#60   Source "/typesense/src/index.cpp", line 1486, in search [0x9001f4]
#50   Source "/typesense/src/index.cpp", line 1788, in search_field [0x8f80f1]
#40   Source "/typesense/src/index.cpp", line 1788, in search_field [0x8f80f1]
#30   Source "/typesense/src/index.cpp", line 1750, in search_field [0x8f6522]
#20   Source "/typesense/src/index.cpp", line 870, in search_candidates [0x8f253e]
#10   Source "/typesense/src/array_base.cpp", line 6, in for_uncompress [0x83b3fb]
#00   Object "/opt/typesense-server", at 0xd16e24, in for_uncompress
Segmentation fault (Address not mapped to object [0xdc000a0060])
```

## Metadata

**Typsense Version**: 0.19.0.rc11 (Docker)

**OS**: macOS 11.1

**Comment by jasonbosco** [comment]
@Nick-Mazuk We fixed one issue in the latest RC build. Could you try this docker image: `typesense/typesense:0.19.0.rc15`

**Comment by Nick-Mazuk** [comment]
Upgrading didn't help. But after some debugging, I was able to figure out that it's happening when using `hidden_hits` with multi-search.

**Comment by jasonbosco** [comment]
I see, this is a different issue then. Could you share the exact search query that's causing the crash?

**Comment by jasonbosco** [comment]
@Nick-Mazuk Would you be able to share a sample dataset that you saw this issue in? I can then try different types of searches and see if I can reproduce the issue, then use the dataset locally to step through the code. 

If so, could you email the sample dataset to contact at typesense dot org

**Comment by Nick-Mazuk** [comment]
Sorry for the delay, @jasonbosco, but I'm sending you the query, dataset, and schema right now via email.

**Comment by kishorenc** [comment]
@Nick-Mazuk Thanks a lot for putting the dataset and query together, but unfortunately I am not able to reproduce it locally against Docker RC15 build. 

1. Are you using any other GET parameters while calling the `GET /multi_search` end-point? 
2. Are there indexing or other operations running against the server when you are querying it?
3. Is the issue consistently reproducible on your machine even after re-indexing the data against a new collection name?

If so, it might be helpful to zip the Typesense data directory and share it -- though I can't think of why it would be any different than importing from outside. Thank you for all the help!

**Comment by Nick-Mazuk** [comment]
@kishorenc, here are the answers to your questions.

> 1\. Are you using any other GET parameters while calling the `GET /multi_search` end-point?

No other query parameters. Headers include the `X-Typesense-API-Key`, and `Content-Type` of `text/plain`. I'm using a POST request because get requests cannot contain a body.


> 2\. Are there indexing or other operations running against the server when you are querying it?

I can reproduce the error without other searching or indexing operations happening at the same time.

> 3\. Is the issue consistently reproducible on your machine even after re-indexing the data against a new collection name?

Perhaps? So most of the data in that dataset is generated with Faker.js, that way I don't need to worry about mixing production data and development data (sure there are other ways to do it, but this seemed easy). I can say that this issue is reproducible after deleting the collection and all its data, then recreating the collection with brand new random data. The collection name has remained the same, though the original collection was deleted. I think this is equivalent to what you're asking?

---

Writing an email to send you the Typesense data directory right now. Do note that I regenerated the dataset since the last email, so I'll also include the updated dataset and query, that way it's consistent with the data in the Typsense data directory. Additionally, I'm working right now to figure out the specific part query that's causing the crash. Hopefully, that will narrow things down and help you replicate it.

**Comment by jasonbosco** [comment]
Just published v0.19.0 with this fix.
